### PR TITLE
UI, support for non-registered users, levels enhancement

### PR DESCRIPTION
**Pull request by caroso1222**
This PR contains the following:

- A good amount of CSS to polish the home screen and the levels page according to the design.
- Support for non-registered users. What I do is to create a dummy user and use this session further. Not sure if it's the right thing, but works. See file 'http-interceptor.ts'
- Enhanced the 'levels' endpoint to return the number of logos and the number of validated so that we can show the progress bar.

### CODE REVIEW

File: apps/api/src/shared/decorators/user.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+/**
+ * Retrieve the logged in user ID from the request context
+ */
+export const CurrentUser = createParamDecorator((data: unknown, context: ExecutionContext) => {
+  const httpContext = context.switchToHttp();
+  const req = httpContext.getRequest();
+  return req.user;
```
Comment (edgargonzalez525) [inline apps/api/src/shared/decorators/user.decorator.ts:9]: where is the user property being injected?

File: apps/logo-quiz/src/app/views/LevelList/components/LevelPreview/LevelPreview.scss [hunk 2]
```diff
@@ -1,7 +1,71 @@
 .level-button {
   display: block;
   border: none;
-  background: lightblue;
-  margin-bottom: 10px;
   min-width: 200px;
+  font-size: 1.25em;
+  font-weight: 600;
+  color: #1D3557;
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/app/views/LevelList/components/LevelPreview/LevelPreview.scss:7]: should we create a file with all colors used as variables?

File: apps/logo-quiz/src/index.html [hunk 3]
```diff
@@ -4,11 +4,19 @@
     <meta charset="utf-8" />
     <title>LogoQuiz</title>
     <base href="/" />
+    <link rel="stylesheet"
+          href="https://cdn.jsdelivr.net/npm/bootstrap-grid-only@1.0.0/bootstrap.min.css">
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/index.html:8]: oh you found a different alternative, good catch

File: apps/logo-quiz/src/main.tsx [hunk 4]
```diff
@@ -8,34 +8,35 @@ import LogoVerify from './app/views/LogoVerify/LogoVerify';
...
       <div>
-        <Route exact path="/" component={App}/>
-        <Route path="/logos/:id" component={LogoVerify}/>
-        <Route path="/levels/:id" component={LogoList}/>
-        <Route exact path="/levels" component={LevelList}/>
-        <Route exact path="/login" component={Login}/>
+        <Route exact path="/" component={App} />
+        <Route path="/logos/:id" component={LogoVerify} />
+        <Route path="/levels/:id" component={LogoList} />
+        <Route exact path="/levels" component={LevelList} />
```
Comment (jdbernalb) [inline apps/logo-quiz/src/main.tsx:36]: There is a typo in the import of this component (LOC: 6)

File: apps/api/src/shared/decorators/user.decorator.ts (hunk 1 above)
Comment (caroso1222) [reply apps/api/src/shared/decorators/user.decorator.ts:9]: It's something Passport does [automatically](https://docs.nestjs.com/techniques/authentication#login-route).

"*Passport automatically creates a user object, based on the value we return from the validate() method, and assigns it to the Request object as req.user*"

File: apps/logo-quiz/src/main.tsx (hunk 4 above)
Comment (caroso1222) [reply apps/logo-quiz/src/main.tsx:37]: Thanks for taking care of that!

File: apps/logo-quiz/src/app/views/LevelList/components/LevelPreview/LevelPreview.scss (hunk 2 above)
Comment (caroso1222) [reply apps/logo-quiz/src/app/views/LevelList/components/LevelPreview/LevelPreview.scss:7]: Yeah. I've yet to figure out how to do that in React, especially how to configure css path mappings as we normally did with Angular's `angular.json`. I'll leave this for later, if that's fine with you, just to move on.

**Pull request merged**